[BoAi01] iP

BoAi01 opened the pull request:
# DukePro

The functionalities include 
* list tasks
* add todos task 
* add events with start and end dates
* add deadlines with end dates
* delete tasks
* mark tasks as done/undone
* search for tasks from keywords

Alternatively, the functionalities include
1. list tasks
2. add todos task 
3. add events with start and end dates
4. add deadlines with end dates
5. delete tasks
6. mark tasks as done/undone
7. search for tasks from keywords

To use a command, simply type the command in the text box, e.g., 
```
todo task
```
will add a `todo` named `task`. :smiley:

Access [this page](https://github.com/BoAi01/ip) for more info about the implementation.

An example of ***italics font***.

This is a blockquote: 
>
>  "We shall not cease from exploration, and the end of all our exploring will be to arrive where we started and know the place for the first time." -- T.S.Eliot
>

This is an example of ~~stikethrough~~.

Commit:
Add Gradle support

BoAi01 committed:
level1 completed

BoAi01 committed:
level 2 completed

BoAi01 committed:
level-3 done

BoAi01 committed:
A-Classes

BoAi01 committed:
Added classes inherited from Task

BoAi01 committed:
Level 4 completed

BoAi01 committed:
added remove task (level 6)

BoAi01 committed:
Level-5. Added exception handling

BoAi01 committed:
Removed redundant exceptions

BoAi01 committed:
Added TextUI testing

BoAi01 committed:
A-Enums

BoAi01 committed:
adjust indent for switch block

BoAi01 committed:
abstract out the command parsing routine to the duke class

BoAi01 committed:
ignore data file

BoAi01 committed:
added save to file

BoAi01 committed:
added command logger

BoAi01 committed:
deleted not needed methods

BoAi01 committed:
added javadoc, cleaned up unused methods

BoAi01 committed:
ignore data file

BoAi01 committed:
change all date printout format to MMM-DD-YYYY

BoAi01 committed:
Revert "change all date printout format to MMM-DD-YYYY"

This reverts commit 154cdc0d719ce1b53e86a1019322ff4d308d23b7.

BoAi01 committed:
Revert "ignore data file"

This reverts commit 5c88f29d6e62cdffb5b5b5bd013fc01df87ac2b8.

BoAi01 committed:
add data file

BoAi01 committed:
add support for date object input and printout

BoAi01 committed:
Merge branch 'branch-Level-7'

BoAi01 committed:
merge with branch-level-8

BoAi01 committed:
adjust text cases after level-8

BoAi01 committed:
More abstractions

BoAi01 committed:
clean up, more javadoc

BoAi01 committed:
organize into packages

BoAi01 committed:
Merge branch 'add-gradle-support' of https://github.com/BoAi01/ip

BoAi01 committed:
modify main class name

BoAi01 committed:
add J-unit test cases

BoAi01 committed:
abstract task type to Task attribute

BoAi01 committed:
fix index error

BoAi01 committed:
Move parser to parser class, add exception

BoAi01 committed:
add java doc

BoAi01 committed:
update Gradle setting

BoAi01 committed:
adjust coding style

BoAi01 committed:
adjusted coding style

BoAi01 committed:
add javadoc

BoAi01 committed:
add find functionality

BoAi01 committed:
Merge branch 'branch-A-JavaDoc'

BoAi01 committed:
Merge branch 'branch-Level-9'

BoAi01 committed:
test commit

BoAi01 committed:
add checkstyle files

BoAi01 committed:
add checkstyle config

BoAi01 committed:
adjust coding style

BoAi01 committed:
remove saving unnecessary commands

BoAi01 committed:
add command classes to remove if-else switches

BoAi01 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/ui/Parser.java

BoAi01 committed:
integrate find functionality

BoAi01 committed:
remove unused methods

BoAi01 committed:
integrate exit command

BoAi01 committed:
remove unused enum

BoAi01 committed:
add number of components check for input commands

BoAi01 committed:
fix a minor style issue

BoAi01 committed:
add record saving

BoAi01 committed:
add completeness check for tasks

BoAi01 committed:
add exception types

BoAi01 committed:
add checking invalid indices and numbers

BoAi01 committed:
modify class names

BoAi01 committed:
add the first prototype of GUI

BoAi01 committed:
add overloaded execute method

BoAi01 committed:
add gui integration

BoAi01 committed:
adjust function names

BoAi01 committed:
delete unnecessarily overloaded methods

BoAi01 committed:
add gui exit

BoAi01 committed:
adjust style based on standard

BoAi01 committed:
add support for searching for multiple strings

EvitanRelta commented on src/main/java/command/DeleteCommand.java:
```diff
@@ -0,0 +1,45 @@
+        String indexToDelete = getCommandContent(command);
+        try {
+            int idx = Integer.parseInt(indexToDelete) - 1;
+            Task t = taskList.get(idx);
```
Should this `t` be renamed to `task` instead to make it clearer? Though, not a code violation.

But, I'm assuming `Task t` is intentionally named as short as possible to follow the ["variables with a small scope can have short names"][link] coding standard?

[link]: https://se-education.org/guides/conventions/java/intermediate.html#:~:text=Variables%20with%20a%20large%20scope%20should%20have%20long%20names%2C%20variables%20with%20a%20small%20scope%20can%20have%20short%20names.

EvitanRelta commented on src/main/java/command/TaskCommand.java:
```diff
@@ -0,0 +1,83 @@
+     */
+    public TaskCommand(String command, boolean doesPrint, int numComponents) throws DukeException {
+        super(command, doesPrint, numComponents);
+        this.commandName = captalizeFirstChar(command.split(" ")[0]);
```
Should these setting of properties be written without the `this` keyword? 

According to the _(advanced)_ ["Avoid unnecessary use of this with fields"][link] coding standard. _(its under the advanced coding standard, which is optional so no need to worry)_

[link]: https://se-education.org/guides/conventions/java/index.html#:~:text=Avoid%20unnecessary%20use%20of%20this%20with%20fields.

EvitanRelta commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,174 @@
+        this.taskList = new TaskList();
+        this.storage = new Storage(RECORD_DIR, RECORD_NAME);
+        this.commandList = new ArrayList<>();
+        this.hasExited = false;
```
Should these setting of properties be written without the `this` keyword? 

According to the _(advanced)_ ["Avoid unnecessary use of this with fields"][link] coding standard. _(its under the advanced coding standard, which is optional so no need to worry)_

[link]: https://se-education.org/guides/conventions/java/index.html#:~:text=Avoid%20unnecessary%20use%20of%20this%20with%20fields.

lxz333 replied:
I think using this.XXX is acceptable in the constructor?

EvitanRelta commented on src/main/java/command/Command.java:
```diff
@@ -0,0 +1,92 @@
+public abstract class Command {
+    protected final String command;
+    protected final boolean doesPrint;
+    protected boolean isExit = false;
```
I like how the booleans have boolean-like names :^)

lxz333 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,174 @@
+    }
+
+    /**
+     * Runs command and handles exceptions
```
Perhaps can consider adding an empty line between the description and @param.

lxz333 commented on src/main/java/command/Command.java:
```diff
@@ -0,0 +1,92 @@
+ */
+public abstract class Command {
+    protected final String command;
+    protected final boolean doesPrint;
```
The naming of doesPrint feels a bit weird to me. Maybe can switch to another expression? (not sure about this

lxz333 reviewed (commented):
Overall the code is clean and clear. I didn't seem much that's improper. I like the way how you split different classes:)

BoAi01 committed:
adjust classname for runShadow

BoAi01 committed:
add assertions

BoAi01 committed:
adjust coding style

BoAi01 committed:
adjust access modifiers of static fields

BoAi01 committed:
replace for loop with stream operation

BoAi01 committed:
Merge pull request #1 from BoAi01/branch-A-Assertions

add assertions

BoAi01 committed:
Merge branch 'master' into branch-A-CodeQuality

BoAi01 committed:
Merge pull request #2 from BoAi01/branch-A-CodeQuality

Branch a code quality

BoAi01 committed:
Merge pull request #3 from BoAi01/branch-A-Streams

replace for loop with stream operation

BoAi01 committed:
Improve code quality

There are some methods not using assertions or in good coding
style.

We use streams operations to replace some for loop to make the
code neater. Further, we replace magic strings and numbers with
constants to improve readability. Assertions are added to some
places to check program flow at runtime.

BoAi01 committed:
add yml for CI

BoAi01 committed:
add partial string search

BoAi01 committed:
add assertion on working path

BoAi01 committed:
remove unused tests at the moment

BoAi01 committed:
adjust background color for dialogue boxes

BoAi01 committed:
add image border

BoAi01 committed:
abstract away magic variables

BoAi01 committed:
add a new message class that wraps String class

BoAi01 committed:
add different color boxing for error message

BoAi01 committed:
add new profile pics

BoAi01 committed:
change duke name

BoAi01 committed:
apply SLAM principle

BoAi01 committed:
move check bye method to command class

BoAi01 committed:
add more exception types and refine exception handling

BoAi01 committed:
apply the SLAM method

BoAi01 committed:
Add encouraging words for users

BoAi01 committed:
Improve exception handling

BoAi01 committed:
apply SLAP principle

BoAi01 committed:
add test cases for task methods

BoAi01 committed:
change name to get

BoAi01 committed:
add test cases

BoAi01 committed:
more test cases

BoAi01 committed:
add user guide and ui pic

BoAi01 committed:
add config file to change the markdown processsor

BoAi01 committed:
Update _config.yml

BoAi01 committed:
Update _config.yml

BoAi01 committed:
Update README.md

BoAi01 committed:
Update README.md

BoAi01 committed:
Update _config.yml

BoAi01 committed:
fix javadoc errors

BoAi01 committed:
fix parsing error

BoAi01 committed:
add final modifer to all constants

BoAi01 committed:
modify test cases for corner case commands

BoAi01 committed:
update user guide

BoAi01 committed:
Update README.md
